test/e2e/restart_changed_config/restart_changed_config_test.go

```diff
@@ -0,0 +1,176 @@
+	Ω(err3).NotTo(gomega.HaveOccurred())
+	Ω(e).NotTo(gomega.BeNil())
+})
+
+var _ = ginkgo.Describe("PodInRecoveryQueue", func() {
+	ginkgo.It("Pod_Restored_In_Recovery_Queue", func() {
+		ginkgo.By("Deploy 1st sleep pod to the dev namespace")
+		podConf, podErr := k8s.InitSleepPod(sleepPodConfigs)
+		Ω(podErr).NotTo(gomega.HaveOccurred())
+		podDev, err := kClient.CreatePod(podConf, dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodRunning(dev, podDev.Name, 1*time.Minute)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By("Deploy 2nd sleep pod to the test namespace")
+		podConf, podErr = k8s.InitSleepPod(sleepPod2Configs)
+		Ω(podErr).NotTo(gomega.HaveOccurred())
+		var podTest *v1.Pod
+		podTest, err = kClient.CreatePod(podConf, test)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodRunning(test, podTest.Name, 1*time.Minute)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "fifo", func(sc *configs.SchedulerConfig) error {
+			// remove placement rules: only allow dev queue
+			sc.Partitions[0].PlacementRules = []configs.PlacementRule{{
+				Name:   types.Tag,
+				Value:  "namespace",
+				Create: false,
+			}}
+
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name:   dev,
+				Parent: false,
+			}); err != nil {
+				return err
+			}
+			return nil
+		})
+
+		ginkgo.By("Check pod in the dev namespace")
+		var appsInfo *dao.ApplicationDAOInfo
+		appsInfo, err = restClient.GetAppInfo("default", "root."+dev, podDev.ObjectMeta.Labels["applicationId"])
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		gomega.Ω(appsInfo).NotTo(gomega.BeNil())
+
+		ginkgo.By("Check pod in the test namespace: recovery queue")
+		var appsInfo2 *dao.ApplicationDAOInfo
+		appsInfo2, err = restClient.GetAppInfo("default", "root."+test, podTest.ObjectMeta.Labels["applicationId"])
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		gomega.Ω(appsInfo2).NotTo(gomega.BeNil())
+
+		ginkgo.By("Restart the scheduler pod")
+		yunikorn.RestartYunikorn(&kClient)
+		ginkgo.By("Port-forward scheduler pod after restart")
+		yunikorn.RestorePortForwarding(&kClient)
+
+		// in case of the plugin startup takes some time, wait until we see the partition
+		ginkgo.By("Check shim registration")
+		err = restClient.WaitForRegistration("default", 5)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Check pod in the dev namespace")
+		appsInfo, err = restClient.GetAppInfo("default", "root."+dev, podDev.ObjectMeta.Labels["applicationId"])
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		gomega.Ω(appsInfo).NotTo(gomega.BeNil())
+
+		ginkgo.By("Check pod in the test namespace: recovery queue")
+		appsInfo2, err = restClient.GetAppInfo("default", "", podTest.ObjectMeta.Labels["applicationId"])
```

Comment:
That is interesting. We can't get app from the `root@recovery@` but it works with partition-level request. not sure why we disallow users from fetching apps from recovery queue directly.

Reply:
there is already a jira for that: https://issues.apache.org/jira/browse/YUNIKORN-2518